tests/unit/test_virtual_serialize.cc

```diff
@@ -611,6 +611,28 @@ INSTANTIATE_TYPED_TEST_CASE_P(
   test_virtual_serialize_inst, TestVirtualSerialize, ConstructTypes,
 );
 
+/*
+ * Test for deserialization when using the base class type
+ */
+
+using TestDeserializationFromBase = TestHarness;
+
+template<typename Base, typename Derived>
+void testDeserializationFromBase() {
+  std::unique_ptr<Base> task(new Derived(TEST_CONSTRUCT{}));
+  auto ret = checkpoint::serialize<Base>(*task);
+  auto out = checkpoint::deserialize<Base>(std::move(ret));
+
+  EXPECT_TRUE(nullptr != out);
```

Comment:
There needs to be some explicit test here, rather than within `check()`, that the deserialization really did produce an instance of `Derived` - either that `dynamic_cast` returns non-`nullptr`, or that `getID()` returns the right enumerator value. Otherwise, the call `out->check()` could dispatch to the `Base` implementation, and mistakenly pass the test.

Reply:
It looks like the new `test_polymorphic.cc` probably covers this concern. I still think this test should be made comprehensive, though, so as not to accept integration of an incomplete test.

Reply:
Right! I added a check for the id returned by `getID()`